core-api/src/test/java/com/optimizely/ab/config/ProjectConfigTest.java

```diff
@@ -345,4 +358,53 @@ public void setForcedVariationMultipleUsers() {
         assertNull(projectConfig.getForcedVariation("etag2", "testUser2"));
 
     }
+
+    @Test
+    public void getAttributeIDWhenAttributeKeyIsEqualtoBotFiltering() {
+        ProjectConfig projectConfig = ProjectConfigTestUtils.validProjectConfigV4();
+        String attributeID = projectConfig.getAttributeId(projectConfig, ReservedAttributeKey.BOT_FILTERING_ATTRIBUTE.toString());
+        assertNull(attributeID);
+    }
+
+    @Test
+    public void getAttributeIDWhenAttributeKeyIsFromAttributeKeyMapping() {
+        ProjectConfig projectConfig = ProjectConfigTestUtils.validProjectConfigV4();
+
+        String attributeID = projectConfig.getAttributeId(projectConfig, "house");
+        assertEquals(attributeID, "553339214");
+    }
+
+    @Test
+    public void getAttributeIDWhenAttributeKeyIsUsingReservedKey() {
+        ProjectConfig projectConfig = ProjectConfigTestUtils.validProjectConfigV4();
+        String attributeID = projectConfig.getAttributeId(projectConfig, "house");
+        assertEquals(attributeID, "553339214");
+    }
+
+    @Test
+    public void getAttributeIDWhenAttributeKeyHaveFromReservedKey() {
```

Comment:
nit. Name is confusing HaveFrom. Thora change krlo. 
And same comment in this one . Use hardcoded useragent string